flax/traverse_util.py

```diff
@@ -49,6 +49,71 @@
 import jax
 
 
+def flatten_dict(xs):
+  """Flatten a nested dictionary.
+
+  The nested keys are flattened to a tuple.
+
+  Example::
+
+    xs = {'foo': 1, 'bar': {'a': 2, 'b': {}}}
+    flat_xs = flatten_dict(xs)
+    print(flat_xs)
+    # {
+    #   ('foo',): 1,
+    #   ('bar', 'a'): 2,
+    # }
+
+  Args:
+    xs: a nested dictionary
+  Returns:
+    The flattened dictionary.
+  """
+  def _flatten(xs, prefix):
+    if not isinstance(xs, dict):
+      return {prefix: xs}
+    result = {}
+    for key, value in xs.items():
+      path = prefix + (key,)
+      result.update(_flatten(value, path))
+    return result
+  return _flatten(xs, ())
```

Comment:
I think it'd be good to assert that `xs` is a dict here -- otherwise you get `flatten_dict(3) == {(): 3}` which seems surprising?

Reply:
agreed